File: architecture/springboot/01-springboot-architecture-layered/src/main/java/com/ganatan/starter/continents/ContinentController.java
package com.ganatan.starter.continents;

import java.util.List;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/continents")
public class ContinentController {

  private final ContinentService service;

  public ContinentController(ContinentService service) {
    this.service = service;
  }

  @PostMapping
  public void add(@RequestBody Continent t) {
    service.add(t);
  }

  @GetMapping
  public List<Continent> list() {
    return service.list();
  }

  @GetMapping("/{id}")
  public Continent get(@PathVariable int id) {
    return service.get(id);
  }

  @PutMapping("/{id}")
  public boolean update(@PathVariable int id, @RequestBody Continent t) {
    return service.update(id, t);
  }

  @DeleteMapping("/{id}")
  public boolean delete(@PathVariable int id) {
    return service.delete(id);
  }
}

File: nestjs/05-nest-continent-module/src/modules/continent/continent.controller.ts
import { Controller, Get } from '@nestjs/common';
import { ContinentService } from './continent.service';

@Controller('continents')
export class ContinentController {
  constructor(private readonly continentService: ContinentService) {}

  @Get()
  getAll() {
    return this.continentService.getAll();
  }
}